plugins/orphans.py

```diff
@@ -0,0 +1,71 @@
+# orphans.py
+# DNF plugin for listing installed packages not required by any other
+# installed package.
+#
+# Copyright (C) 2015 Emil Renner Berthing
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+from collections import deque
+
+import dnf
+import dnf.sack
+import dnf.cli
+
+
+class Orphans(dnf.Plugin):
+    name = 'Orphans'
+
+    def __init__(self, base, cli):
+        super(Orphans, self).__init__(base, cli)
+        if cli:
+            cli.register_command(OrphansCommand)
+
+
+class OrphansCommand(dnf.cli.Command):
+    aliases = ('orphans',)
+    summary = 'List installed packages not required by any other package'
+
+    def run(self, args):
+        sack = dnf.sack.rpmdb_sack(self.base)
+        allpackages = set(sack.query())
+
+        depends = dict()
+        orphans = allpackages.copy()
+        for pkg in allpackages:
+            deps = set()
+            for req in pkg.requires:
+                sreq = str(req)
+                if sreq.startswith('rpmlib(') or sreq == 'solvable:prereqmarker':
+                    continue
+                providers = frozenset(sack.query().filter(provides=req))
+                if pkg not in providers:
+                    deps.update(providers)
+            depends[pkg] = deps
+            orphans.difference_update(deps)
+
+        queue = deque()
+        for pkg in orphans:
+            print(pkg)
```

Comment:
I agree all around except this. This is the whole point of the plugin. That is listing the installed packages which are not required by any other installed package.
I use this feature of pacman all the time, `pacman -Qt`, but with all other package managers you need external tools like `package-cleanup --all --leaves` (which by the way has a bug), `deborphan -anp1`, `rpmorphan -all`, etc.
The reason this list is useful is that it gives you a nice overview of what is installed on your system. If you have anything you don't need it'll be on this list (otherwise it would be needed as a dependency), and conversely if you want to uninstall anything (without breaking dependencies) it must involve at least one of the packages on this list.